.github/workflows/docs-build.yml

```
@@ -0,0 +1,19 @@
+name: docs-build
+
+on:
+  push:
+    branches:
+      - main
+  pull_request_target: ~
```

Being explicit here is preferred and I don't think we need `ready_for_review` and `converted_to_draft` events to trigger.

```suggestion
  pull_request_target:
    types:
      - opened
      - reopened
      - synchronize
```

---

This is ok.

The default is `opened`, `reopened` and `synchronize` anyway if no type is specified.

> By default, a workflow only runs when a pull_request_target event's activity type is opened, synchronize, or reopened.

Source: https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request_target